### PR TITLE
gnc_list_length comparator functions

### PR DESCRIPTION
**Pull request by christopherlam**
for review candidate g_list_length comparator function names... `gnc_list_length_gt` `gnc_list_length_lt` `gnc_list_length_eq`?

### CODE REVIEW

File: libgnucash/core-utils/gnc-glib-utils.c [hunk 1]
```diff
@@ -351,3 +351,33 @@ gnc_g_list_stringjoin (GList *list_of_strings, const gchar *sep)
 
     return retval;
 }
+
+gboolean
+gnc_list_length_cmp (GList *lst, size_t len,
+                     gboolean less, gboolean same, gboolean more)
```
Comment (jralls) [inline libgnucash/core-utils/gnc-glib-utils.c:357]: That's ugly API that shouldn't be public, so `static inline boolean list_length_cmp_internal`.

**Comment by jralls** [comment]
That will work, but what's the use case?

**Comment by christopherlam** [comment]
These would be used when undoing b5f5129f7.

**Comment by jralls** [comment]
You only need `gnc_list_length_gt` for that, and the intent is less clear than `gnc_list_length_at_least` (which could be renamed `gnc_list_length_ge`. The indirection of the (very ugly) `list_length_cmp_internal` gains nothing: There's already `g_list_length` and `!gnc_list_length_at_least` is equivalent to `gnc_list_length_lt`.

**Comment by jralls** [comment]
It occurs to me that you could change the cmp function to be like traditional c-library compares such as strcmp and return -1 for less, 0 for equal, and 1 for greater. That would get rid of the ugly boolean inputs. OTOH the only benefit to testing equality is if length is much less than g_list_length as in that case it would cut short the list traversal; the same is also true of less-than. In both cases one would get the benefit only if one expected the answer to be false in the majority of instances.

**Comment by christopherlam** [comment]
In this case `cmp` will be the only function required. Callers can compare the `gint` result as desired, from the whole <= < == > => range.

**Comment by jralls** [comment]
Good.

**Pull request merged**